packages/app/cypress/e2e/integration/runs.spec.ts

```diff
@@ -1,80 +1,249 @@
+        if (obj.result.data?.cloudProjectsBySlugs) {
+          for (const proj of obj.result.data.cloudProjectsBySlugs) {
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="no-runs"]')
     })
 
-    cy.loginUser()
-    cy.visitApp()
+    it('displays the projectId as it is in their config file', () => {
+      cy.withCtx(async (ctx) => {
+        await ctx.actions.file.writeFileInProject('cypress.config.js', 'module.exports = {\'projectId\': \'abcdef\'}')
+      })
+
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        if (obj.result.data?.cloudProjectsBySlugs) {
+          for (const proj of obj.result.data.cloudProjectsBySlugs) {
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('projectId: \'abcdef\'')
+    })
+
+    it('displays correct source control instructions', () => {
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        if (obj.result.data?.cloudProjectsBySlugs) {
+          for (const proj of obj.result.data.cloudProjectsBySlugs) {
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Ensure that your cypress.config.js file is checked into source control:')
+      cy.get('[data-cy="copy-button"]').first().click()
+      cy.contains('Copied!')
+    })
 
-    cy.get('[href="#/runs"]').click()
-    cy.contains('Connect your project').should('exist')
+    it('displays correct instructions on how to record a run', () => {
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        if (obj.result.data?.cloudProjectsBySlugs) {
+          for (const proj of obj.result.data.cloudProjectsBySlugs) {
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Run this command in your local development terminal or in CI:')
+      cy.contains('--record --key 2aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa')
+      cy.get('[data-cy="copy-button"]').last().click()
+      cy.contains('Copied!')
+    })
   })
 
-  it('if logged in and connected', { viewportWidth: 1200 }, () => {
-    cy.loginUser()
-    cy.visitApp()
-    cy.intercept('mutation-ExternalLink_OpenExternal', { 'data': { 'openExternal': true } }).as('OpenExternal')
+  context('Runs - Runs List', () => {
+    it('if a run has been recorded, a list of recorded runs is displayed', () => {
```

Comment:
Nitpicking here for consistency
```suggestion
    it('displays a list of recorded runs if a run has been recorded', () => {
```